### PR TITLE
Basic unit test for shares and CLI flag for scanning

### PR DESCRIPTION
**Pull request by mathiascode**
Partial fix for https://github.com/Nicotine-Plus/nicotine-plus/issues/443. The rescan flag is not documented yet, since it doesn't work well if Nicotine+ is already open (for now, you will have to quit Nicotine+, run `nicotine --rescan` and wait for it to complete, then start Nicotine+ again).

Also added some unit tests for shares.

### CODE REVIEW

File: debian/rules [hunk 1]
```diff
@@ -35,9 +35,10 @@ SOURCE_DIR  = $(abspath $(PACKAGE_DIR)/../)
 #  embedded directly in the Python source...
 VERSION_UPSTREAM_DISTUTILS = $(shell sed -n 's/^version\s*=\s*"\(\S*\)"$$/\1/p' $(SOURCE_DIR)/pynicotine/utils.py)
 
+# Disable shares test, since shares are not saved for some reason...
```
Comment (mathiascode) [inline debian/rules:38]: @kiplingw Do you have any idea why this fails in autopkgtest? The dbs/shelves for shares are seemingly empty after a scan.

Comment (mathiascode) [reply debian/rules:38]: Seems to work when the tests run the second time, in debian/tests/control.

Comment (kiplingw) [reply debian/rules:38]: No idea. Where is the log? Possibly a race condition.

**Comment by kiplingw** [comment]
Fails in [Groovy](https://launchpadlibrarian.net/500431509/buildlog_ubuntu-groovy-amd64.nicotine_2.1.2-202010030202~ubuntu20.10.1_BUILDING.txt.gz). At least I think this is the guilty PR.

**Pull request merged**